### PR TITLE
Update debugger

### PR DESCRIPTION
**Pull request by s123unny**
- Add VS Code extension
- Add test cases
- Add support for local variables

### CODE REVIEW

File: aeneas/src/ir/Normalization.v3 [hunk 1]
```diff
@@ -653,7 +653,6 @@ class ReachabilityNormalizer(config: NormalizerConfig, ra: ReachabilityAnalyzer)
 			if (specializer.normCode(context, rm)) return;
 		}
 		SsaRaNormalizer.new(context, this).build(rm.norm);
-		if (rm.norm.source != null) rm.norm.source.irIndex = newIr.methods.length;
```
Comment (titzer) [inline aeneas/src/ir/Normalization.v3:656]: I'm not sure why this line was removed. Can you explain?

File: aeneas/src/ssa/SsaBuilder.v3 [hunk 2]
```diff
@@ -291,6 +291,19 @@ class SsaBuilder extends SsaBlockState {
 		// if (SsaCheckpoint.?(block.prev)) return;
 		if (CLOptions.DEBUG.get()) block.append(SsaCheckpoint.new(source, vals));
 	}
+	def addNewVar(name: string, vtype: Type, index: int, vals: Array<SsaInstr>) -> SsaInstr {
+		if (Debug.PARANOID) { checkInputs(vals); }
+		if (end || CLOptions.TARGET.get() != null) return null; // TODO
```
Comment (titzer) [inline aeneas/src/ssa/SsaBuilder.v3:296]: Hmm, this should probably check whether the debugger extension is active?

File: aeneas/src/ir/Normalization.v3 (hunk 1 above)
Comment (s123unny) [reply aeneas/src/ir/Normalization.v3:656]: I add this field `irIndex` before to try to retrieve irMethod from vstMethod in debugger. It is not used anymore

File: aeneas/src/ssa/SsaDebugger.v3 [hunk 3]
```diff
@@ -36,14 +38,51 @@ enum DbgDisplay(cmdName: string, alias: string, help: string) {
...
+
+enum DbgStopReason {
+	stopOnBreakpoint,
+	stopOnStep,
+	stopOnEntry,
+	end,
+}
+
+class DbgFrame {
+	def localVars = Vector<(string, Type, int, SsaInstr)>.new(); // name, index, instr
```
Comment (titzer) [inline aeneas/src/ssa/SsaDebugger.v3:82]: You can introduce:

```
type DbgFrameEntry(name: string, t: Type, index: int, instr: SsaInstr) #unboxed { }
```

**Pull request merged**